File: solutions/clojure/atbash-cipher/1/src/atbash_cipher.clj
(ns atbash-cipher
  (:require [clojure.string :as str]))

(def ^:private cipher
  (zipmap "abcdefghijklmnopqrstuvwxyz0123456789"
          "zyxwvutsrqponmlkjihgfedcba0123456789"))
(defn- translate [text]
  (->> (str/replace (str/lower-case text) #"[^a-zA-Z0-9]" "")
       (map cipher)
       (apply str)))

(defn encode [plaintext]
  (->> (translate plaintext)
       (partition-all 5)
       (map str/join)
       (str/join " ")))

(def decode translate)

File: solutions/ruby/atbash-cipher/1/atbash_cipher.rb
# frozen_string_literal: true

# module Atbash
module Atbash
  A_Z = 'abcdefghijklmnopqrstuvwxyz'
  Z_A = A_Z.reverse
  def self.encode(plaintext) =
    plaintext
      .downcase
      .gsub(/\W/, '')
      .tr(A_Z, Z_A)
      .scan(/.{1,5}/)
      .join(' ')

  def self.decode(ciphertext) = encode(ciphertext).delete(' ')
end
